### PR TITLE
[TorchMhlo] Add permute and transpose

### PR DESCRIPTION
**Pull request by fortianyou**

### CODE REVIEW

File: pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/MhloLegalizeUtils.cpp [hunk 1]
```diff
@@ -25,6 +25,17 @@
 namespace mlir {
 namespace mhlo {
 
+std::vector<int64_t> normalizeDimIndex(ArrayRef<int64_t> dims, int64_t rank) {
```
Comment (wyzero) [inline pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/MhloLegalizeUtils.cpp:28]: I see this function defined in #416  as well. Should we just remove one?

Comment (fortianyou) [reply pytorch_blade/src/torch-mlir/lib/Conversion/TorchToMhlo/MhloLegalizeUtils.cpp:28]: I will rebase it later.

File: pytorch_blade/tests/disc/ops/test_permutation.py [hunk 2]
```diff
@@ -41,9 +46,10 @@ def test_aten_t(self):
...
             return x.t()
 
-        self._test_permute(transpose, x=torch.tensor(5, device=self.device))
-        self._test_permute(transpose, x=torch.randn([], device=self.device))
-        self._test_permute(transpose, x=torch.randn([5], device=self.device))
+        if not isTorchMlirEnable():
+            self._test_permute(transpose, x=torch.tensor(5, device=self.device))
+            self._test_permute(transpose, x=torch.randn([], device=self.device))
+            self._test_permute(transpose, x=torch.randn([5], device=self.device))
         self._test_permute(transpose, x=torch.randn([3, 4], device=self.device))
```
[the remark this answers is not in the document]
Comment (fortianyou) [reply pytorch_blade/tests/disc/ops/test_permutation.py:53]: I will add comments on this in later PR.

**Review by Yancey1989 — APPROVED** [review]
LGTM

**Pull request merged**